File: backend/FloriculturaEmbeleze/FloriculturaEmbeleze.Domain/Entities/Product.cs
namespace FloriculturaEmbeleze.Domain.Entities;

public class Product : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal BasePrice { get; set; }
    public int StockQuantity { get; set; } = 0;
    public bool IsFeatured { get; set; }
    public bool IsActive { get; set; } = true;
    public string? Badge { get; set; }

    public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();
    public ICollection<ProductVariation> Variations { get; set; } = new List<ProductVariation>();
    public ICollection<ProductCategory> ProductCategories { get; set; } = new List<ProductCategory>();
    public ICollection<StockMovement> StockMovements { get; set; } = new List<StockMovement>();
    public ICollection<ProductSupplier> ProductSuppliers { get; set; } = new List<ProductSupplier>();
    public ICollection<ProductPromotion> ProductPromotions { get; set; } = new List<ProductPromotion>();
}

File: frontend/src/types/product.ts
export interface Product {
  id: string
  name: string
  slug: string
  description: string | null
  basePrice: number
  stockQuantity: number
  isFeatured: boolean
  isActive: boolean
  badge: string | null
  mainImageUrl: string | null
  categoryNames: string[]
  createdAt: string
  updatedAt: string | null
}

export interface ProductDetail extends Product {
  images: ProductImage[]
  variations: ProductVariation[]
  categories: CategoryRef[]
}

export interface ProductImage {
  id: string
  imageUrl: string
  displayOrder: number
  isMain: boolean
}

export interface ProductVariation {
  id: string
  name: string
  price: number
  stockQuantity: number
  isActive: boolean
}

export interface CategoryRef {
  id: string
  name: string
  slug: string
}

export interface ProductCreate {
  name: string
  description: string | null
  basePrice: number
  stockQuantity: number
  isFeatured: boolean
  isActive: boolean
  badge: string | null
  categoryIds: string[]
  variations: ProductVariationCreate[]
}

export interface ProductVariationCreate {
  name: string
  price: number
  stockQuantity: number
}

export interface ProductFilter {
  search?: string
  categoryId?: string
  minPrice?: number
  maxPrice?: number
  sortBy?: string
  page?: number
  pageSize?: number
  isActive?: boolean
}

export interface PaginatedResult<T> {
  items: T[]
  totalCount: number
  page: number
  pageSize: number
  totalPages: number
}
